src/cpp/server/DBActiveSessionStorage.cpp

```diff
@@ -0,0 +1,404 @@
+   }
+
+   return Success();
+}
+
+Error DBActiveSessionStorage::getConnectionOrOverride(boost::shared_ptr<database::IConnection>* connection)
+{
+   if(overrideConnection_ == nullptr)
+      return getConn(connection);
+   else
+   {
+      *connection = overrideConnection_;
+      return Success();
+   }
+}
+
+DBActiveSessionStorage::DBActiveSessionStorage(const std::string& sessionId)
+   : sessionId_(sessionId)
+{
+   
+}
+
+DBActiveSessionStorage::DBActiveSessionStorage(const std::string& sessionId, boost::shared_ptr<core::database::IConnection> overrideConnection)
+   : sessionId_(sessionId), overrideConnection_(overrideConnection)
+{
+
+}
+
+Error DBActiveSessionStorage::readProperty(const std::string& name, std::string* pValue)
+{
+   *pValue = "";
+   boost::shared_ptr<database::IConnection> connection;
+   Error error = getConnectionOrOverride(&connection);
+   
+   if(error)
+   {
+      return error;
+   }
+
+
+   database::Query query = connection->query("SELECT "+name+" FROM "+kTableName+" WHERE "+kSessionIdColumnName+" = :id")
+      .withInput(sessionId_);
+
+   database::Rowset results{};
+   error = connection->execute(query, results);
+
+   if(!error)
+   {
+      database::RowsetIterator iter = results.begin();
+      if(iter != results.end())
+      {
+         if(name != kUserId)
+         {
+            *pValue = iter->get<std::string>(name, "");
+         }
+         else
+         {
+            *pValue = std::to_string(iter->get<int>(name));
+         }
+         if(++iter != results.end()) 
+         {
+            return Error{"Too many sessions returned", errc::TooManySessionsReturned, ERROR_LOCATION};
+         }
+      }
+      else
+      {
+         return Error{"Session does not exist", errc::SessionNotFound, ERROR_LOCATION};
+      }
+   }
+   else
+   {
+      return Error{"DatabaseException", errc::DBError, "Database exception during property read [ session:"+sessionId_+" property:"+name+" ]", error, ERROR_LOCATION};
+   }
+   
+   return error;
+}
+
+Error DBActiveSessionStorage::readProperties(const std::set<std::string>& names, std::map<std::string, std::string>* pValues)
+{
+   pValues->clear();
+   boost::shared_ptr<database::IConnection> connection;
+   Error error = getConnectionOrOverride(&connection);
+
+   if(error)
+   {
+      return error;
+   }
+   
+   std::string namesString = getColumnNameList(names);
+   database::Query query = connection->query("SELECT "+namesString+" FROM "+kTableName+" WHERE "+kSessionIdColumnName+"=:id")
+      .withInput(sessionId_);
+
+   database::Rowset rowset{};
+   error = connection->execute(query, rowset);
+
+   if(!error)
+   {
+      database::RowsetIterator iter = rowset.begin();
+      if(iter != rowset.end())
+      {
+         populateMapWithRow(iter, pValues);
+         // Sanity check number of returned rows, by using the pk in the where clause we should only get 1 row
+         if(++iter != rowset.end())
```

Comment:
There should be spaces between the `if` and the `(`. Could you add throughout?
```suggestion
         if (++iter != rowset.end())
```